modules/apps/frontend-js/frontend-js-react-web/types/src/main/resources/META-INF/resources/js/hooks/useEventListener.d.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+/**
```

Comment:
Let's see how this goes, but maybe Brian might ask for an empty line here? 

What I was going to ask, though was if `types` is a strong enough signal that these are autogenerated or if we could make it even more obvious, similarly to what you are doing in the `tsconfig.json` file... 🤔

Reply:
> but maybe Brian might ask for an empty line here?

He might. But if he does, I am hoping that this answer will suffice: "These are autogenerated files created by a tool that we don't control (`tsc`, the TypeScript compiler), so we have our source formatting — Prettier and ESLint in this case — configured to ignore them).

> What I was going to ask, though was if `types` is a strong enough signal that these are autogenerated or if we could make it even more obvious, similarly to what you are doing in the `tsconfig.json` file...

There's nothing that we can change _inside_ the files (eg. adding a `@generated` header) because TS will overwrite them any time we build. Maybe, just maybe, we could add a post-processing step to add our own stuff there, but then you wouldn't be able to enjoy `yarn run tsc --watch`, which is pretty useful... well, you could still run it, but we're in the situation of our tool fighting uselessly against a third-party tool again. An easier change would be to make the directory name more obvious (`__generated__`, `generated-types`... whatever... Heck, we might even be able to make `.generated/types/` stick, or something).

If you have any bright ideas we can implement them now, or just try this and see what Brian says.